### DISCUSSION TITLE
Image optimizations

### DISCUSSION THREAD

**Discussion by MaxKellermann** [Openvario/meta-openvario#275] [Show and tell]
Quick benchmark of the OpenVario optimizations I made in the past weeks:

| Version | Compressed image | rootfs | Cold boot to menu |
| --- | ---: | ---: | ---: |
| [20193](https://github.com/Openvario/meta-openvario/releases/download/20193/OpenVario-linux-openvario-image-glibc-ipk-20193-openvario-7-PQ070.rootfs.img.gz) | 85 MB | 194 MB | 27.5s |
| 2da58cb9d94c62fb310e940100b6957969a12fd1 | 43 MB | 90 MB | 12.5s |
| ed6ded58ce24650bd3724a6707fc0b3c8a1bde5d | 38 MB | 82 MB |
| experimental | 33 MB | 70 MB | 10.5s |

All optimizations were made without losing features.

My experimental build (last line) includes unfinished patches which I havn't yet submitted; some of the changes:
- no variod (I'd like to use XCSoar's audio vario instead)
- no PulseAudio (was only necessary for variod)
- sensord will also be removed (I'd like to integrate that into XCSoar as well)
- no connman (using systemd-networkd & wpa_supplicant instead)
- no ovmenu-ng (replaced with [XCSoar-based menu](https://github.com/Openvario/meta-openvario/pull/199))
- no bash (was only needed by ovmenu-ng)

**Comment by kedder** [comment]
Please keep connman!

**Reply by MaxKellermann** [reply]
I will not submit the connman-removal patch for merging until the new GUI menu has full support for wpa_supplicant.

If you wish to keep connman, declare a RDEPENDS (or better: RRECOMMENDS) on it from whatever package needs it; for example, your Python menu should have that dependency, because it's a frontend for connman. That way, those who install your menu will automatically have connman. But it makes no sense to make connman a top-level image dependency, because it image itself has no use for connman.

**Reply by kedder** [reply]
Users do use connman. It is part of the feature set. Just like everything else. You can just as well implement a full support of connman in your GUI menu for that matter.

**Reply by MaxKellermann** [reply]
No, I won't implement connman support, because connman is just an overly complex frontend for wpa_supplicant. Why talk to connman, when talking to wpa_supplicant is much simpler? I have a wpa_supplicant client + UI already, I just need to wire it into the OpenVarioMenu.
